test/service_manager/service_manager.go

```diff
@@ -0,0 +1,156 @@
+package service_manager
+
+import (
+	"github.com/Shopify/sarama"
+	"github.com/strimzi/strimzi-canary/internal/config"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// Implementation of Service Manager
+type ServiceManager struct {
+	CanaryConfig
+	Paths
+}
+
+// Configurations of canary that is manipulated in e2e tests
+type CanaryConfig struct {
+	ReconcileIntervalTime string
+	TopicTestName         string
+	KafkaBrokerAddress    string
+
+}
+
+// paths to services ( kafka zookeeper docker compose, and Canary application main method)
+type Paths struct {
+	pathDockerComposeKafkaZookeeper string
+	pathToCanaryMain string
+}
+
+const (
+	canaryTestTopicName         = "__strimzi_canary_test_topic"
+	kafkaBrokerAddress          = "127.0.0.1:9092"
+	canaryReconcileIntervalTime = "1000"
+
+	pathToDockerComposeImage    = "compose-kafka-zookeeper.yaml"
+	pathToMainMethod            = "../cmd/main.go"
+)
+
+func (c *ServiceManager) StartKafkaZookeeperContainers() {
+	log.Println("Starting kafka & Zookeeper")
+
+	var cmd = exec.Command("docker-compose", "-f", c.pathDockerComposeKafkaZookeeper, "up", "-d" )
+	if err := cmd.Run(); err != nil {
+		log.Fatal(err)
+	}
+	log.Println("Zookeeper and Kafka containers created")
+	// after creation of containers we still have to wait for some time before successful communication with Kafka & Zookeper
+	c.waitForBroker()
```

Comment:
the canary already has a retry mechanism with exponential backoff for connecting to the Kafka cluster, do we really need another try with another client creation with some attempts and sleep?

Reply:
Because I am unable to write some sort of "@BeforeClass" and in some of tests i need to be able to create Sarama Admin (which I am not immediately after creation of Broker). 
In summary waiting has nothing to do with Canary but with need to perform all tests correctly.  Possible solution is move it to tests but as number of those which will require it will be only bigger ...